src/components/WalletSelector/WalletSelector.tsx

```diff
@@ -34,13 +64,32 @@ function WalletSelector() {
   const { hideModal } = useModal();
 
   useEffect(() => {
-    if (account && !isLoggedIn && signer) {
-      signMessageAndAuthenticate(account, signer).then((jwt) => {
-        logIn(jwt);
-        hideModal();
-      });
+    if (account && isConnecting && !isLoggedIn && signer) {
```

Comment:
added `isConnecting` to this condition so that we only automatically trigger `signMessageAndAuthenticate` if the user has clicked a wallet option in the Wallet modal to set `isConnecting` to true.